src/compute/src/sink/copy_to_s3_oneshot.rs

```diff
@@ -67,68 +81,272 @@ where
 }
 
 fn copy_to<G>(
```

Comment:
From a maintainability perspective, I am wondering if it is possible to extract functions to reduce the length of this one? (optional)